org.idempiere.test/src/org/idempiere/test/costing/NonStockedExpTypeAvgPOCostingTest.java

```diff
@@ -0,0 +1,771 @@
+import org.compiere.model.MProductPrice;
+import org.compiere.model.MProduction;
+import org.compiere.model.MProductionLine;
+import org.compiere.model.MRMA;
+import org.compiere.model.MRMALine;
+import org.compiere.model.MStorageOnHand;
+import org.compiere.model.ProductCost;
+import org.compiere.model.Query;
+import org.compiere.model.SystemIDs;
+import org.compiere.process.DocAction;
+import org.compiere.process.DocumentEngine;
+import org.compiere.process.ProcessInfo;
+import org.compiere.process.ServerProcessCtl;
+import org.compiere.util.Env;
+import org.compiere.util.TimeUtil;
+import org.compiere.util.Util;
+import org.compiere.wf.MWorkflow;
+import org.eevolution.model.MPPProductBOM;
+import org.eevolution.model.MPPProductBOMLine;
+import org.idempiere.test.AbstractTestCase;
+import org.idempiere.test.DictionaryIDs;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Isolated;
+
+/**
+ * Test cases Stocked Expense type product with Average PO Costing Method.
+ * 
+ * @author Logilite Technologies
+ * @since  September 15, 2023
+ */
+@Isolated
+public class NonStockedExpTypeAvgPOCostingTest extends AbstractTestCase
+{
+	private static final BigDecimal	BD_2	= new BigDecimal("2.00");
+	private static final BigDecimal	BD_5	= new BigDecimal("5.00");
+	private static final BigDecimal	BD_20	= new BigDecimal("20.00");
+
+	public NonStockedExpTypeAvgPOCostingTest()
+	{
+	}
+
+	@Test
+	public void testMaterialReceipt()
+	{
+		//need to create category and product for the every test case because after the first test completed it's rollback 
+		//and also need to delete category and product at the end of the each test case otherwise next test case failed
+		MProductCategory category = createProductCategory();
+		MProduct product = createProduct(category.get_ID());
+
+		try
+		{
+			MAcctSchema as = getAccountingSchema();
+			MCost cost = getCost(as, product);
+
+			// Create Material Receipt
+			MInOutLine rLine = createPOAndMRForProduct(product);
+
+			MMatchPO[] matchPOs = MMatchPO.get(Env.getCtx(), rLine.getM_InOutLine_ID(), getTrxName());
+			assertEquals(1, matchPOs.length);
+			assertEquals(rLine.getC_OrderLine_ID(), matchPOs[0].getC_OrderLine_ID());
+			assertEquals(rLine.getM_InOutLine_ID(), matchPOs[0].getM_InOutLine_ID());
+			assertTrue(matchPOs[0].getQty().compareTo(BigDecimal.TEN) == 0);
+			
+			cost.load(getTrxName());
+			
+			//Validate that averagePO costing method should not be created in the cost for the expense type product
+			validateCostingForAveragePO(product.get_ID());
+
+		
+			// Testing cost and stock of the product after completing MR
+			assertEquals(BD_2, cost.getCurrentCostPrice().setScale(2, RoundingMode.HALF_UP), "Unexpected current cost price");
+			
+			assertEquals(BigDecimal.TEN.setScale(2, RoundingMode.HALF_UP), cost.getCurrentQty().setScale(2, RoundingMode.HALF_UP), "Unexpected current Qty");
+			// Test to check that storage should nor created for the expense type product
```

Comment:
// Assert no storage records created for expense type product

Reply:
@hengsin  line 123 has that assertions.

Reply:
I means change the comment text, not the code

Reply:
ohk, Corrected and pushed changes.

Reply:
Resolved